internal/apps/devflow/flow/flow.service.go

```diff
@@ -234,19 +236,70 @@ func (s *Service) OperationMerge(ctx context.Context, req *pb.OperationMergeRequ
 		if err != nil {
 			return nil, err
 		}
-		return &pb.OperationMergeResponse{}, nil
+	} else {
+		// Is already removed
+		if !mrInfo.IsJoinTempBranch {
+			return &pb.OperationMergeResponse{}, nil
+		}
+		err = s.RejoinTempBranch(ctx, tempBranch, mrInfo.TargetBranch, sourceBranch, appDto)
+		if err != nil {
+			return nil, err
+		}
 	}
 
-	// Is already removed
-	if !mrInfo.IsJoinTempBranch {
-		return &pb.OperationMergeResponse{}, nil
+	userID := apis.GetUserID(ctx)
+	if err := s.CreateFlowEvent(&CreateFlowRequest{
+		ProjectID: appDto.ProjectID,
+		AppID:     appDto.ID,
+		OrgID:     appDto.OrgID,
+		Data: pb.FlowEventData{
+			IssueID:          issueID,
+			Operator:         userID,
+			TempBranch:       tempBranch,
+			SourceBranch:     sourceBranch,
+			TargetBranch:     mrInfo.TargetBranch,
+			AppName:          appDto.DisplayName,
+			IsJoinTempBranch: strconv.FormatBool(req.Enable.Value),
+		},
+	}); err != nil {
+		s.p.Log.Errorf("failed to create flow event, err: %v", err)
 	}
-	err = s.RejoinTempBranch(ctx, tempBranch, mrInfo.TargetBranch, sourceBranch, appDto)
+
+	return &pb.OperationMergeResponse{}, nil
+}
+
+type CreateFlowRequest struct {
+	Data      pb.FlowEventData
```

Comment:
Use pointer is better, because `DoNotCopy` pragma is embedded in pb MessageState

Reply:
done